### PR TITLE
Fix dLOOP tests and improve DLoop core logic

### PR DESCRIPTION
**Pull request by dinosaurchi**
### Overview
Comprehensive test suite overhaul for dLOOP vault system - a leveraged ERC-4626 vault enabling fixed-ratio exposure to collateral assets with automated debt management.

### Key Fixes
- **26+ test files** fixed across core, periphery, and mock components
- **Critical bugs resolved**:
  - Double-counting collateral bug in `DLoopIncreaseLeverageBase` (DoS fix)
  - Borrow shortfall protection against lending pool rounding errors
  - Inflation attack mitigation for first deposits
- **Enhanced coverage**: Edge cases, error conditions, regression tests
- **Deployment improvements**: Separated CoreLogic deployment, fixed dependencies

### Impact
- Security vulnerabilities eliminated
- Test reliability improved  
- Better maintainability and debugging
- 46 files changed (+1,629 / -3,328 lines)

### Validation
All tests pass, ensuring dLOOP system robustness and security.

### CODE REVIEW

**Comment by dinosaurchi** [comment]
All DLoop related contracts were deployed successfully with:
```shell
make deploy
```

Log snippet:
```txt
...
Deploying DLoopCoreLogic on network hardhat (chainId: 31337)
Deploying DLoopCoreLogic library contract...
deploying "DLoopCoreLogic" (tx: 0x93da763a1853186bee518c1c3acfcf4a56e1c18a3c6d3edfcb2055bba9dea107)...: deployed at 0x3904b8f5b0F49cD206b7d5AABeE5D1F37eE15D8d with 545850 gas
DLoopCoreLogic library deployed successfully
Deploying dLOOP core vaults on network hardhat (chainId: 31337)
Deploying dLOOP core vault: 3x_sFRAX_dUSD
Deploying DLoopCoreDLend-3X-sfrxUSD with parameters:
  name: dLOOP 3X sfrxUSD dLEND
  symbol: 3X-sfrxUSD
  collateralToken: 0xDc64a140Aa3E981100a9becA4E685f962f0cF6C9
  debtToken: 0x04C89607413713Ec9775E14b954286519d836FEf
  lendingPoolAddressesProvider: 0x40918Ba7f132E0aCba2CE4de4c4baF9BD2D7D849
  targetLeverageBps: 3000000
  lowerBoundTargetLeverageBps: 2000000
  upperBoundTargetLeverageBps: 4000000
  maxSubsidyBps: 20000
  minDeviationBps: 20000
  withdrawalFeeBps: 4000
  rewardsController: 0xab8146260506618d388e1c8A750c13d0245F39fC
  dLendAssetToClaimFor: 0xe4cd1C9448E210d0B6430b604816dC6Db2BaeC67
  targetStaticATokenWrapper: 0x71089Ba41e478702e1904692385Be3972B2cBf9e
  treasury: 0x70997970C51812dc3A010C7d01b50e0d17dc79C8
  maxTreasuryFeeBps: 1000
  initialTreasuryFeeBps: 500
  initialExchangeThreshold: 100
deploying "DLoopCoreDLend-3X-sfrxUSD" (tx: 0xba6f69851e4318efa1b7b9d04da702c48bdb83cf67f8a3a377e606e90fbb3557)...: deployed at 0x2Dd78Fd9B8F40659Af32eF98555B8b31bC97A351 with 4962902 gas
All dLOOP core vaults deployed successfully
Deploying DLoopQuoter on network hardhat (chainId: 31337)
Deploying DLoopQuoter contract...
deploying "DLoopQuoter" (tx: 0xbf84cb91f47b4e0eba55a785eb16083b8b4fb9a40a81dc71358497d688fe5d7e)...: deployed at 0x56fC17a65ccFEC6B7ad0aDe9BD9416CB365B9BE8 with 377936 gas
DLoopQuoter deployed successfully
Deploying Odos swap logic on network hardhat (chainId: 31337)
deploying "OdosSwapLogic" (tx: 0x8b6483a23971a91646791f45c52feb7c45a2c2670cab5fa59598339d9b609254)...: deployed at 0x2625760C4A8e8101801D3a48eE64B2bEA42f1E96 with 503222 gas
Odos swap logic deployed successfully
Deploying Odos depositor on network hardhat (chainId: 31337)
deploying "DLoopDepositorOdos" (tx: 0x54c81342fc858562160d9d78070285325aef01aea33afbc81a455098846ad02b)...: deployed at 0xFE5f411481565fbF70D8D33D992C78196E014b90 with 1621278 gas
Odos depositor deployed successfully
Deploying Odos redeemer on network hardhat (chainId: 31337)
deploying "DLoopRedeemerOdos" (tx: 0x02b7aaaabb12ba8f30ce8d2ac8bf33646e9f887d4b52a83fcda9f11f9c36301d)...: deployed at 0xD6b040736e948621c5b6E0a494473c47a6113eA8 with 1714890 gas
Odos redeemer deployed successfully
Deploying Odos decrease leverage on network hardhat (chainId: 31337)
deploying "DLoopDecreaseLeverageOdos" (tx: 0x5f61548dcd4e517e7fccebc605d7fa84e777e4f5f29c689cb07794e24b31a5b3)...: deployed at 0x139e1D41943ee15dDe4DF876f9d0E7F85e26660A with 1639803 gas
Odos decrease leverage deployed successfully
Deploying Odos increase leverage on network hardhat (chainId: 31337)
deploying "DLoopIncreaseLeverageOdos" (tx: 0x235ecd293885c1eacdb1d77616562c7efc4fbef992fe0ff9905027a9afe0c0ed)...: deployed at 0xAdE429ba898c34722e722415D722A70a297cE3a2 with 1747379 gas
Odos increase leverage deployed successfully
...
```

All tests are passing
```shell
make test
```

Output snippet
```txt
...
  1370 passing (40s)
  131 pending

✨  Done in 43.58s.
yarn run v1.22.19
$ /Users/dinosaurchi/Desktop/Project/stably-prime/trinity/sonic-solidity-contracts-2/node_modules/.bin/jest --detectOpenHandles '--testPathPattern=test\.unit\.ts' --passWithNoTests
No tests found, exiting with code 0
✨  Done in 0.78s.
yarn run v1.22.19
$ /Users/dinosaurchi/Desktop/Project/stably-prime/trinity/sonic-solidity-contracts-2/node_modules/.bin/jest --detectOpenHandles '--testPathPattern=test\.integ\.ts' --passWithNoTests
No tests found, exiting with code 0
✨  Done in 0.53s.
```

All DLoop tests are passing
```shell
yarn hardhat test test/dloop/**/*.ts
```

Output snippet
```
...
  539 passing (28s)

✨  Done in 31.86s.
```

**Review by Git-on-my-level — APPROVED** [review]

File: typescript/common/bps_constants.ts [hunk 1]
```diff
@@ -2,3 +2,4 @@
 export const ONE_BPS_UNIT = 100; // 1 bps with 2 decimals
 export const ONE_PERCENT_BPS = 100 * ONE_BPS_UNIT; // 1% in basis points
 export const ONE_HUNDRED_PERCENT_BPS = 100 * ONE_PERCENT_BPS; // 100% in basis points
+export const TARGET_LEVERAGE_BPS = 300 * ONE_PERCENT_BPS; // 3x leverage (300%)
```
[the remark this answers is not in the document]
Comment (dinosaurchi) [reply typescript/common/bps_constants.ts]: Removed https://github.com/dtrinity/sonic-solidity-contracts/pull/134/commits/704051c9f1b601fac2c9f91265e2d13b161decd1
